component/core/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/LabelDAO.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * This file is part of the Meeds project (https://meeds.io/).
+ *
+ * Copyright (C) 2023 Meeds Association contact@meeds.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.exoplatform.social.core.jpa.storage.dao.jpa;
+
+import org.exoplatform.commons.persistence.impl.GenericDAOJPAImpl;
+import org.exoplatform.social.core.jpa.storage.entity.LabelEntity;
+
+import javax.persistence.NoResultException;
+import javax.persistence.TypedQuery;
+import java.util.ArrayList;
+import java.util.List;
+
+public class LabelDAO extends GenericDAOJPAImpl<LabelEntity, Long> {
+  public LabelEntity findLabelByObjectTypeAndObjectIdAndLang(String objectType, String objectId, String language) {
+    TypedQuery<LabelEntity> query = getEntityManager()
+            .createNamedQuery("SocLabelEntity.findLabelByObjectTypeAndObjectIdAndLang",
+                    LabelEntity.class)
+            .setParameter("objectType", objectType)
+            .setParameter("objectId", objectId)
+            .setParameter("language", language);
+    try {
+      return query.getSingleResult();
+    } catch (NoResultException e) {
+      return null;
+    }
+
+  }
+
+  public List<LabelEntity> findLabelByObjectTypeAndObjectId(String objectType, String objectId) {
+    TypedQuery<LabelEntity> query = getEntityManager()
+            .createNamedQuery("SocLabelEntity.findLabelByObjectTypeAndObjectId",
+                    LabelEntity.class)
+            .setParameter("objectType", objectType)
+            .setParameter("objectId", objectId);
+    try {
+      return query.getResultList();
+    } catch (NoResultException e) {
+      return new ArrayList<>();
+    }
```

Comment:
The NoResultException can't be thrown by `getResultList()`
![Capture d’écran du 2023-02-08 13-35-46](https://user-images.githubusercontent.com/594824/217531649-6417f392-c4b3-4183-9a37-d49a6024b156.png)

It can be thrown using `getSingleResult()` only
![Capture d’écran du 2023-02-08 13-36-06](https://user-images.githubusercontent.com/594824/217531804-26aa88a2-0022-4e7b-b6e9-adfef3ae0a12.png)

( Please update other methods as well )

```suggestion
    return query.getResultList();
```

Reply:
done